custom_components/octopus_energy/heat_pump/sensor_live_cop.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
+import logging
+from typing import List
+
+from homeassistant.const import (
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+    UnitOfPower
+)
+from homeassistant.core import HomeAssistant, callback
+
+from homeassistant.util.dt import (now)
+from homeassistant.helpers.update_coordinator import (
+  CoordinatorEntity
+)
+from homeassistant.components.sensor import (
+  RestoreSensor,
+  SensorDeviceClass,
+  SensorStateClass,
+)
+
+from .base import (BaseOctopusEnergyHeatPumpSensor)
+from ..utils.attributes import dict_to_typed_dict
+from ..api_client.heat_pump import HeatPump
+from ..coordinators.heat_pump_configuration_and_status import HeatPumpCoordinatorResult
+
+_LOGGER = logging.getLogger(__name__)
+
+class OctopusEnergyHeatPumpSensorLiveCoP(CoordinatorEntity, BaseOctopusEnergyHeatPumpSensor, RestoreSensor):
+  """Sensor for displaying the live CoP of a heat pump."""
+
+  def __init__(self, hass: HomeAssistant, coordinator, heat_pump_id: str, heat_pump: HeatPump):
+    """Init sensor."""
+    # Pass coordinator to base class
+    CoordinatorEntity.__init__(self, coordinator)
+    BaseOctopusEnergyHeatPumpSensor.__init__(self, hass, heat_pump_id, heat_pump)
+
+    self._state = None
+    self._last_updated = None
+
+  @property
+  def unique_id(self):
+    """The id of the sensor."""
+    return f"octopus_energy_heat_pump_{self._heat_pump_id}_live_cop"
+
+  @property
+  def name(self):
+    """Name of the sensor."""
+    return f"Live CoP ({self._heat_pump_id})"
```

Comment:
suggestion: I'm a bit nervous of calling this live due to the API calls being delayed in nature (at least for the moment until local access is available), as it could be a minute old (or longer if the call fails). It should probably follow the convention of the electricity meter sensors being either

`Latest CoP Heat Pump ({self._heat_pump_id})`

or 

`Current CoP Heat Pump ({self._heat_pump_id})`

Other namings (e.g. class, unique_id) should be updated accordingly

Reply:
I chose to mirror the API, terminology can be very misleading here as this is effecitvely an "instantaneous coeffiecent of performance", rather than a lookback over any period of time, so I can appreciate why they've chosen to name it "live" (despite it not being live in the API either!). My only other suggestion would be "Instant". The staleness of data is an existing behaviour of Home Assistant, and I'm not sure "Latest" or "Current" do much to solve for that. Thoughts?

Reply:
I'm falling towards "current" for consistency, but lets keep it how it is for consistency with the API instead. We should clarify in the docs the nature of the data